internal/client.go

```diff
@@ -1003,6 +1027,32 @@ func DialCloudOperationsClient(ctx context.Context, options CloudOperationsClien
 	}, nil
 }
 
+// PrepareUpdateWorkflowOperation returns an UpdateWorkflowOperation that can be used to perform Update-with-Start,
+// or an error in case the update operation is invalid.
+func PrepareUpdateWorkflowOperation(options UpdateWorkflowOptions) (*UpdateWorkflowOperation, error) {
+	input, err := createUpdateWorkflowInput(options)
+	if err != nil {
+		return nil, err
+	}
+	return &UpdateWorkflowOperation{input: input}, nil
+}
+
+// Get returns a WorkflowUpdateHandle, or an error if the workflow run is invalid.
+func (op *UpdateWorkflowOperation) Get(run WorkflowRun) (WorkflowUpdateHandle, error) {
```

Comment:
I'm going back and forth on this API. Options I'm considering:

**(1) `Get(run WorkflowRun) (WorkflowUpdateHandle, error)`** (as written here)

The safest but most verbose option. The `run` parameter is used to ensure the function is invoked _after_ `ExecuteWorkflow` and that the operation matches the workflow run. But it requires an extra error check that is _only_ there to prevent a mistake by the user, not to surface an actual runtime error.

**(2) `Get(run WorkflowRun) WorkflowUpdateHandle`**

Removes the error check. If the user passes in `nil`, it would panic (akin to a NPE). If the user passes in the wrong workflow run, it would panic (not ideal).

**(3) `Get(ctx context.Context, valuePtr interface{}) error`**

If the operation embeds `WorkflowUpdateHandle`, the extra step to obtain the handle is removed entirely. While this is the most convenient option, it also means that there is a risk the user invokes the handle's functions before executing the operation (if `Get` is invoked too early, it would return the appropriate error).

**(4) `GetResult(ctx context.Context, run WorkflowRun, valuePtr interface{}) error`**

Combined with (1), adding an additional shortcut to get the update outcome directly compresses two calls into one. This is based on the assumption that in most cases, users care about the result more than anything else from the `WorkflowUpdateHandle` (but they could still obtain it, if needed).

Essentially it comes down to safety vs convenience. Since all users would benefit from less verbose code, I think the question is how likely the user errors are the safest options would prevent/make clearer.